src/middlewared/middlewared/plugins/interface/configure.py

```diff
@@ -33,19 +33,17 @@ def configure(self, data, aliases, wait_dhcp=False, options=None):
 
         has_ipv6 = data['int_ipv6auto'] or False
 
-        if (
-            not self.middleware.call_sync('system.is_freenas') and
-            self.middleware.call_sync('failover.node') == 'B'
-        ):
-            ipv4_field = 'int_ipv4address_b'
-            ipv6_field = 'int_ipv6address'
-            alias_ipv4_field = 'alias_v4address_b'
-            alias_ipv6_field = 'alias_v6address_b'
-        else:
-            ipv4_field = 'int_ipv4address'
-            ipv6_field = 'int_ipv6address'
-            alias_ipv4_field = 'alias_v4address'
-            alias_ipv6_field = 'alias_v6address'
+        if self.middleware.call_sync('failover.licensed'):
```

Comment:
This is a change of behavior, we dont need license to know if node is A and B, I am skeptical about regressions if we change this because we need to do this based on hardware alone, and not if its licensed.

Reply:
apologies, glanced over that when I was replacing all the `is_freenas` calls and didn't realize it.